File: source/rolx-client/src/app/reports/core/report-range-type.ts
import { assertDefined } from '@app/core/util/utils';

export class ReportRangeType {
  id!: string;
  label!: string;
  hasCustomStart!: boolean;
  hasCustomEnd!: boolean;
  hasCustomEndMonth!: boolean;
  validateModel() {
    assertDefined(this, 'id');
    assertDefined(this, 'label');
    assertDefined(this, 'hasCustomStart');
    assertDefined(this, 'hasCustomEnd');
    assertDefined(this, 'hasCustomEndMonth');
  }
}

File: source/RolXServer/RolXServer/Reports/WebApi/Resource/ReportRangeType.cs
// -----------------------------------------------------------------------
// <copyright file="ReportRangeType.cs" company="Christian Ewald">
// Copyright (c) Christian Ewald. All rights reserved.
// Licensed under the MIT license.
// See LICENSE.md in the project root for full license information.
// </copyright>
// -----------------------------------------------------------------------

namespace RolXServer.Reports.WebApi.Resource;

/// <summary>
/// The type of a reports range.
/// </summary>
public record ReportRangeType(string Id, string Label, bool HasCustomStart, bool HasCustomEnd, bool HasCustomEndMonth);
